internal/backend/grpc/router/talos_backend.go

```diff
@@ -107,17 +107,22 @@ func (l *TalosBackend) GetConnection(ctx context.Context, fullMethodName string)
 	}
 
 	// overwrite the node headers with the resolved ones
+
 	resolved := resolveNodes(l.nodeResolver, md)
-	if resolved.node.Address != "" {
-		md = md.Copy()
 
-		setHeaderData(ctx, md, "node", resolved.node.Address)
-	} else if len(resolved.nodes) > 0 {
+	if len(resolved) > 0 {
+		headerKey := "node"
```

Comment:
idk if this is going to be a problem, but it seems to rewrite `node`/`nodes` based on result count?

I don't quite know all the details, but would it be easier to write something like:

```go
if node, ok := md.Get("node"); ok {
    md.Set("node", resolveNodes(node))
}

if nodes, ok := md.Get("nodes"); ok {
    md.Set("node", resolveNodes(nodes)...)
}
```

Reply:
That wouldn't work exactly because of the code structure, but I did something similar to that.